oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/method/annotation/OAuth2AuthorizedClientArgumentResolver.java

```diff
@@ -101,12 +103,12 @@ public Object resolveArgument(MethodParameter parameter,
 		if (principal == null) {
```

Comment:
If anonymous Authentication is disabled, this might be null so we shouldn't throw an Exception here.

Reply:
If the user is using `RegisteredOAuth2AuthorizedClient` and has configured `HttpSecurity.anonymous().disabled()` than IMO this is a mis-configuration and the `IllegalStateException` signals this to the user. Does this make sense?

Reply:
Why is this a misconfiguration? Is there any reason this should not work if the principal is null?

Reply:
`OAuth2AuthorizedClientService` requires `Principal.getName()` in order to load/save an `OAuth2AuthorizedClient` so if `HttpSecurity.anonymous().disabled()` is configured than `Authentication` will be `null` and `RegisteredOAuth2AuthorizedClient` can either return null (which you are suggesting) or throw the exception as current behaviour. 

If we don't throw the exception to signal to the user (and return `null` instead) than they might think there is a bug because it's not resolving the `OAuth2AuthorizedClient`. At least with the exception and detailed message, the user will than know to either `HttpSecurity.anonymous().enable()` or remove the use of `@RegisteredOAuth2AuthorizedClient`.